### PR TITLE
parallel: add v20240622

### PR DESCRIPTION
**Pull request by drkrynstrng**
Adding latest GNU Parallel release version 20240622.

### CODE REVIEW

**Comment by drkrynstrng** [comment]
@spackbot re-run pipeline

**Review by alalazo — APPROVED** [review]

**Review by alalazo — CHANGES_REQUESTED** [review]
There are legit issues in pipelines:
- https://gitlab.spack.io/spack/spack/-/jobs/11501806
- https://gitlab.spack.io/spack/spack/-/jobs/11501808
- https://gitlab.spack.io/spack/spack/-/jobs/11501807

**Comment by drkrynstrng** [comment]
Okay, I'll look into the failed jobs.

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/parallel/package.py [hunk 1]
```diff
@@ -34,6 +35,7 @@ def check(self):
         # The Makefile has a 'test' target, but it does not work
         make("check")
 
+    depends_on("gmake", type="build")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/parallel/package.py]: This is inherited from `AutotoolsPackage`:
```suggestion
```

**Review by alalazo — CHANGES_REQUESTED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by drkrynstrng** [comment]
For some reason it's failing to install in the aws-isc CI jobs. Based on the error messages, it looks like parallel is being run as a shell script instead of a Perl script when generating the Bash completions, but I haven't figured out why yet.

**Pull request closed without merging**